### DISCUSSION TITLE
Selecting a Cache Size

### DISCUSSION THREAD

**Discussion by MrCowKing** [DNSCrypt/dnscrypt-proxy#2223] [Q&A] [answered]
Is there a soft upper limit that the size of the cache should be, or is the limit more along the lines of "what ever the hardware is able to hold in memory and the number of requests the processor can respond to"? For a relatively modern (4C/8T & 8GB RAM) computer being used as a server for various services, would increasing the cache size to 16,384 or 32,768 be reasonable?

**Comment by jumpsmm7** [comment, marked as answer]
> Is there a soft upper limit that the size of the cache should be, or is the limit more along the lines of "what ever the hardware is able to hold in memory and the number of requests the processor can respond to"? For a relatively modern (4C/8T & 8GB RAM) computer being used as a server for various services, would increasing the cache size to 16,384 or 32,768 be reasonable?

The only way to actually know if you should increase the size of the cache would be to see if the cache ever gets reasonably full before caches entries start to expire and drop off. For this you would probably need some type of builtin dnscrypt proxy cache analysis/debug command. I honestly don't believe it has one of those...
